calculator.rb

```diff
@@ -0,0 +1,186 @@
+# if any of these three condition isn't met, then num_1 and num_2 can be formatted as they are with the other operations
+# must check if dividing by zero is a problem after determining that both are integers, but before calculating modulo
+# this is bc modulo can't be calculated if num_2 is zero as it will make the program throw an error message
+# format_userinput_for_division formats the numbers, performs the calculation, and returns the calculation
+# format_userinput_not_including_division formats the numbers, but doesn't perform the calculation
+def format_userinput_for_division(operator, valid_operators, num_1, num_2, user_input)
+  if num_1 =~ /^(\d)+$/ && num_2 =~ /^(\d)+$/
+    unless divide_by_zero(num_2.to_i)
+      exit
+    end
+    modulo = (num_1.to_i % num_2.to_i)
+    if modulo != 0
+      result = (num_1.to_f / num_2.to_f)
+      return result
+    else
+      format_userinput_not_including_division(valid_operators, num_1, num_2, user_input)
+      result = division(user_input[:num_1], user_input[:num_2])
+      return result
+    end
+  else
+    format_userinput_not_including_division(valid_operators, num_1, num_2, user_input)
+    result = division(user_input[:num_1], user_input[:num_2])
+    return result
+  end
+end
+
+# format user input for addition, subtration, and multiplication, but not for division
+# only formats the numbers, does not perform the actual calculation
+def format_userinput_not_including_division(valid_operators, num_1, num_2, user_input)
+  if num_1 =~ /^(\d)*\.(\d)+$/
+    num_1 = num_1.to_f
+    user_input[:num_1] = num_1
+  elsif num_1 =~ /^(\d)+$/
+    num_1 = num_1.to_i
+    user_input[:num_1] = num_1
+  else
+    puts "Whoa nelly! It looks like you entered something other than a number."
+    exit
+  end
+  if num_2 =~ /^(\d)*\.(\d)+$/
+    num_2 = num_2.to_f
+    user_input[:num_2] = num_2
+  elsif num_2 =~ /^(\d)+$/
+    num_2 = num_2.to_i
+    user_input[:num_2] = num_2
+  else
+    puts "Whoa nelly! It looks like you entered something other than a number."
+    exit
+  end
+end
+
+# if the result of a calculation is zero, display 0 never 0.0
+def format_if_result_is_zero(result)
+  if result == 0 || result == 0.0
+    result = 0
+  end
+  return result
+end
+
+##### END FUNCTIONS #####
+
+##### START BODY #####
+
+# display welcome message
+puts "Welcome to The Handy Dandy Calculator!\n\n"
+
+# request operation, num_1, and num_2 from the user
+request_operations_numbers(user_input, valid_operators)
+
+# if performing addition, format the user input, perform the calculation, format result if it's zero, display equation and final result
+if valid_operators[:+].include?(user_input[:operation])
+  format_userinput_not_including_division(valid_operators, user_input[:num_1], user_input[:num_2], user_input)
+  result = addition(user_input[:num_1], user_input[:num_2])
+  result = format_if_result_is_zero(result)
+  puts "#{user_input[:num_1]} + #{user_input[:num_2]} = #{result}"
```

Comment:
These last two lines (calling `format_if_result_is_zero` and `puts`ing) are almost the same for every operation. Could you eliminate this redundancy somehow?